src/utils.ts

```diff
@@ -303,3 +313,49 @@ export async function createUnsignedFillRelayTransaction(
     { from: simulatedRelayerAddress }
   );
 }
+
+/**
+ * Search back in time for the first block at or older than the desired lookback.
+ *
+ * The approach here is as below:
+ * 1. We use the skip distances, which are meant to be a couple of hours or less for each chain.
+ * 2. Fetch the first block one skip distance from the latest block. Check its timestamp to see how far back it is
+ * relatively to the latest block.
+ * 3. Apply a multiplier to the skip distance based on the time difference and find a block with the desired lookback
```

Comment:
This step 3 toook me a while to understand but I think it might be better summarized as. "3. Use the implied block speed from the latest block to one skip distance earlier to approximate how many blocks earlier we need to lookback to find a block older than the target block. This assumes that the block speed is roughly constant from latest block until the target block."

Reply:
Updated